File: src/main/java/com/smartassembly/backend/entity/RegistrationRequest.java
package com.smartassembly.backend.entity;

import com.smartassembly.backend.enums.RegistrationStatus;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import java.time.LocalDate;

import java.time.LocalDateTime;

@Entity
@Table(name = "registration_requests")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RegistrationRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "request_id") // Проверь, как называется колонка в БД. Если просто "id", убери эту строку
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "assembly_id", nullable = false)
    private Assembly assembly;

    @Column(name = "first_name", nullable = false)
    private String firstName;

    @Column(name = "last_name", nullable = false)
    private String lastName;

    @Column(name = "middle_name")
    private String middleName;

    @Column(name = "email", nullable = false)
    private String email;

    @Column(name = "phone", nullable = false)
    private String phone;

    @Enumerated(EnumType.STRING)
    @Column(name = "status")
    @Builder.Default // Чтобы builder использовал значение по умолчанию
    private RegistrationStatus status = RegistrationStatus.PENDING;

    @Column(name = "birth_date")
    private LocalDate birthDate;

    @Column(name = "iin")
    private String iin;

    @Column(name = "instagram")
    private String instagram;

    @Column(name = "study_place")
    private String studyPlace;

    @Column(name = "work_place")
    private String workPlace;

    @Column(name = "volunteering_experience", columnDefinition = "TEXT")
    private String volunteeringExperience;

    @Column(name = "hobbies", columnDefinition = "TEXT")
    private String hobbies;

    @Column(name = "free_days")
    private String freeDays;

    @Column(name = "languages")
    private String languages;

    @Column(name = "photo_url")
    private String photoUrl;

    @Column(name = "interested_events", columnDefinition = "TEXT")
    private String interestedEvents;

    @Column(name = "discovery_source")
    private String discoverySource;

    @Column(name = "motivation", columnDefinition = "TEXT")
    private String motivation;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reviewed_by")
    private User reviewedBy;

    @Column(name = "reviewed_at")
    private LocalDateTime reviewedAt;

    @Column(name = "hr_comment")
    private String hrComment;

    @OneToOne
    @JoinColumn(name = "created_user_id")
    private User createdUser;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
}

File: mobile/lib/models/registration_request.dart
class RegistrationRequest {
  final int id;
  final String firstName;
  final String? middleName;
  final String lastName;
  final String phone;
  final String email;
  final String? instagram;
  final String? assemblyName;
  final String? photoUrl;
  final DateTime? birthDate;
  final String? iin;
  final String? studyPlace;
  final String? workPlace;
  final String? freeDays;
  final String? languages;
  final String? volunteeringExperience;
  final String? hobbies;
  final String? interestedEvents;
  final String? discoverySource;
  final String? motivation;
  final String status;
  final String? hrComment;
  final DateTime? createdAt;
  final DateTime? reviewedAt;

  RegistrationRequest({
    required this.id,
    required this.firstName,
    this.middleName,
    required this.lastName,
    required this.phone,
    required this.email,
    this.instagram,
    this.assemblyName,
    this.photoUrl,
    this.birthDate,
    this.iin,
    this.studyPlace,
    this.workPlace,
    this.freeDays,
    this.languages,
    this.volunteeringExperience,
    this.hobbies,
    this.interestedEvents,
    this.discoverySource,
    this.motivation,
    required this.status,
    this.hrComment,
    this.createdAt,
    this.reviewedAt,
  });

  factory RegistrationRequest.fromJson(Map<String, dynamic> json) {
    DateTime? parseDate(dynamic v) {
      if (v == null) return null;
      try {
        return DateTime.parse(v as String);
      } catch (_) {
        return null;
      }
    }

    return RegistrationRequest(
      id: (json['id'] as num).toInt(),
      firstName: json['firstName'] as String? ?? '',
      middleName: json['middleName'] as String?,
      lastName: json['lastName'] as String? ?? '',
      phone: json['phone'] as String? ?? '',
      email: json['email'] as String? ?? '',
      instagram: json['instagram'] as String?,
      assemblyName: json['assemblyName'] as String?,
      photoUrl: json['photoUrl'] as String?,
      birthDate: parseDate(json['birthDate']),
      iin: json['iin'] as String?,
      studyPlace: json['studyPlace'] as String?,
      workPlace: json['workPlace'] as String?,
      freeDays: json['freeDays'] as String?,
      languages: json['languages'] as String?,
      volunteeringExperience: json['volunteeringExperience'] as String?,
      hobbies: json['hobbies'] as String?,
      interestedEvents: json['interestedEvents'] as String?,
      discoverySource: json['discoverySource'] as String?,
      motivation: json['motivation'] as String?,
      status: json['status'] as String? ?? 'PENDING',
      hrComment: json['hrComment'] as String?,
      createdAt: parseDate(json['createdAt']),
      reviewedAt: parseDate(json['reviewedAt']),
    );
  }

  String get fullName =>
      [lastName, firstName, middleName].where((s) => s != null && s.isNotEmpty).join(' ');

  String get initials {
    final f = firstName.isNotEmpty ? firstName[0] : '';
    final l = lastName.isNotEmpty ? lastName[0] : '';
    return (f + l).toUpperCase();
  }

  String get statusDisplay {
    switch (status) {
      case 'PENDING':
        return 'Ожидает';
      case 'APPROVED':
        return 'Принята';
      case 'REJECTED':
        return 'Отклонена';
      default:
        return status;
    }
  }
}
